File: level2/lovely-lucky-lambs/Solution.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public static int stingy(int total_lambs) {
        List<Integer> stingyList = new ArrayList<>();
        stingyList.add(1);
        stingyList.add(1);
        int x = 2, total = 2;
        while (x <= total_lambs) {
            int value = stingyList.get(x - 1) + stingyList.get(x - 2);
            stingyList.add(value);
            total += value;
            if (total > total_lambs)
                break;
            x++;
        }
        return stingyList.size();
    }

    public static int generous(int total_lambs) {
        List<Integer> generousList = new ArrayList<>();
        int x = 0, total = 0;
        while (x <= total_lambs) {
            int current = (int) Math.pow(2, x);
            generousList.add(current);
            total += current;
            if (total > total_lambs)
                break;
            x++;
        }
        return generousList.size();
    }

    public static int solution(int total_lambs) {
        return stingy(total_lambs) - generous(total_lambs);
    }
}

File: level2/lovely-lucky-lambs/solution.py
def stingy(total_lambs):
    stingy_list = [1, 1]
    x, total = 2, 2
    while x <= total_lambs:
        value = stingy_list[x - 1] + stingy_list[x - 2]
        stingy_list.append(value)
        total += value
        if total > total_lambs:
            break
        x += 1
    return len(stingy_list)

def generous(total_lambs):
    generous_list = []
    x, total = 0, 0
    while x <= total_lambs:
        current = 2 ** x
        generous_list.append(current)
        total += current
        if total > total_lambs:
            break
        x += 1
    return len(generous_list)

def solution(total_lambs):
    return stingy(total_lambs) - generous(total_lambs)
